### PR TITLE
feat: LT-1170 - Link users to sites

### PR DESCRIPTION
**Pull request by jfaracik**
* Add endpoint for assigning user to sites
* Add manager for sites to get users
* Update serializers accordingly
* Bit of code cleanup (eg removing JSONParsers)
* Remove unnecessary comments

### CODE REVIEW

**Comment by a-bek** [comment]
lgtm

File: applications/views/applications.py [hunk 1]
```diff
@@ -57,18 +58,19 @@ def get_queryset(self):
...
+            users_sites = Site.objects.get_by_user_and_organisation(self.request.user, self.request.user.organisation)
+            disallowed_applications = SiteOnApplication.objects.exclude(site__id__in=users_sites).values_list(
+                "application", flat=True
+            )
+            filtered_applications = BaseApplication.objects.exclude(id__in=disallowed_applications).exclude(
+                application_type=ApplicationType.HMRC_QUERY
+            )
+            draft = get_case_status_by_status(CaseStatusEnum.DRAFT)
+
+            if submitted:
```
Comment (Mark-Spiers-kainos) [inline applications/views/applications.py]: if this is done before `.exclude(id__in=disallowed_applications)`  you can use the object manager to get `.submitted()` `.drafts()` which contains the same code.

File: conf/permissions.py [hunk 2]
```diff
@@ -4,12 +4,17 @@
 from users.models import ExporterUser
 
 
-def assert_user_has_permission(user, permission, organisation: Organisation = None):
+def has_permission(user, permission, organisation: Organisation = None):
```
[the remark this answers is not in the document]
Comment (a-bek) [reply conf/permissions.py]: Would it make sense to put this as a method on the user object?

`request.user.has_permission(...)`

File: users/serializers.py [hunk 3]
```diff
@@ -67,10 +68,18 @@ def get_organisations(self, instance):
 
     def get_role(self, instance):
         if self.context:
-            role = UserOrganisationRelationship.objects.get(user=instance, organisation=self.context).role
+            role = get_user_organisation_relationship(instance, self.context).role
             return RoleSerializer(role).data
         return None
 
+    def get_sites(self, instance):
+        from organisations.serializers import SiteViewSerializer
```
[the remark this answers is not in the document]
Comment (jfaracik) [reply users/serializers.py]: Sadly not :(

File: conf/permissions.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jfaracik) [reply conf/permissions.py]: That's a great idea - I've moved it :)

File: users/libraries/get_user.py [hunk 4]
```diff
@@ -58,7 +57,7 @@ def get_users_from_organisation(pk):
         raise NotFoundError({"organisation": "Organisation not found - " + str(pk)})
 
 
-def get_user_organisation_relationship(user: ExporterUser, organisation: Organisation):
+def get_user_organisation_relationship(user, organisation) -> UserOrganisationRelationship:
```
[the remark this answers is not in the document]
Comment (jfaracik) [reply users/libraries/get_user.py]: I've removed it just to be consistent, we should discuss as a team if we want this to be a consistent thing across the codebase.

File: organisations/serializers.py [hunk 5]
```diff
@@ -147,10 +150,26 @@ def create(self, validated_data):
 
 class SiteViewSerializer(serializers.ModelSerializer):
     address = AddressSerializer()
+    users = serializers.SerializerMethodField()
+
+    def get_users(self, instance):
+        users = set([x.user for x in UserOrganisationRelationship.objects.filter(sites__id__exact=instance.id)])
```
[the remark this answers is not in the document]
Comment (jfaracik) [reply organisations/serializers.py]: That returns the ID (not the object we need) sadly

**Pull request merged**